presto-hive/src/main/java/com/facebook/presto/hive/metastore/glue/GlueHiveMetastore.java

```diff
@@ -0,0 +1,703 @@
+    {
+        try {
+            TableInput input = GlueInputConverter.convertTable(table);
+            glueClient.createTable(new CreateTableRequest()
+                    .withDatabaseName(table.getDatabaseName())
+                    .withTableInput(input));
+        }
+        catch (AlreadyExistsException e) {
+            throw new TableAlreadyExistsException(new SchemaTableName(table.getDatabaseName(), table.getTableName()));
+        }
+        catch (EntityNotFoundException e) {
+            throw new SchemaNotFoundException(table.getDatabaseName());
+        }
+        catch (AmazonServiceException e) {
+            throw new PrestoException(HIVE_METASTORE_ERROR, e);
+        }
+    }
+
+    @Override
+    public void dropTable(String databaseName, String tableName, boolean deleteData)
+    {
+        Table tbl = getTableOrElseThrow(databaseName, tableName);
+
+        try {
+            glueClient.deleteTable(new DeleteTableRequest()
+                    .withDatabaseName(databaseName)
+                    .withName(tableName));
+        }
+        catch (AmazonServiceException e) {
+            throw new PrestoException(HIVE_METASTORE_ERROR, e);
+        }
+
+        String tblLocation = tbl.getStorage().getLocation();
+        if (deleteData && !isExternalTable(tbl) && !isNullOrEmpty(tblLocation)) {
+            deleteDir(hdfsContext, hdfsEnvironment, new Path(tblLocation), true);
+        }
+    }
+
+    private static boolean isExternalTable(Table tbl)
+    {
+        return tbl.getTableType().equals(EXTERNAL_TABLE.name());
+    }
+
+    private static void deleteDir(HdfsContext context, HdfsEnvironment hdfsEnvironment, Path path, boolean recursive)
+    {
+        try {
+            hdfsEnvironment.getFileSystem(context, path).delete(path, recursive);
+        }
+        catch (Exception e) {
+            // don't fail if unable to delete path
+            Logger.get(GlueHiveMetastore.class).warn(e, "Failed to delete path: " + path.toString());
+        }
+    }
+
+    @Override
+    public void replaceTable(String databaseName, String tableName, Table newTable, PrincipalPrivileges principalPrivileges)
+    {
+        try {
+            TableInput newTableInput = GlueInputConverter.convertTable(newTable);
+            glueClient.updateTable(new UpdateTableRequest()
+                    .withDatabaseName(databaseName)
+                    .withTableInput(newTableInput));
+        }
+        catch (EntityNotFoundException e) {
+            throw new TableNotFoundException(new SchemaTableName(databaseName, tableName));
+        }
+        catch (AmazonServiceException e) {
+            throw new PrestoException(HIVE_METASTORE_ERROR, e);
+        }
+    }
+
+    @Override
+    public void renameTable(String databaseName, String tableName, String newDatabaseName, String newTableName)
+    {
+        throw new GlueUnsupportedException(HIVE_METASTORE_ERROR, "Table rename is not yet supported by Glue service");
```

Comment:
This should be a `PrestoException(NOT_SUPPORTED)` error, since the operation the user requested is not supported. `HIVE_METASTORE_ERROR` is used when there is a (normally transient) metastore error.

Reply:
Got it, will update these